debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -107,6 +113,45 @@ private PostgresReplicationConnection(Configuration config,
         }
     }
 
+    protected void initPublication() {
+        if (PostgresConnectorConfig.LogicalDecoder.PGOUTPUT.equals(plugin)) {
+            LOGGER.info("Initializing PgOutput logical decoder publication");
+            try {
+                try (Statement stmt = pgConnection().createStatement()) {
+                    ResultSet rs = stmt.executeQuery(String.format(
+                            "SELECT COUNT(1) FROM pg_publication WHERE pubname = '%s'",
+                            publicationName
+                    ));
+                    if (rs.next()) {
+                        Long count = rs.getLong(1);
+                        if (count == 0L) {
+                            LOGGER.info("Creating new publication '{}' for plugin '{}'", publicationName, plugin);
+                            // Publication doesn't exist, create it.
+                            // todo: DBZ-766 - Change this to be restricted based on configured whitelist tables?
+                            //      For situations where no publication exists, we likely cannot create it for all tables.
+                            //      This is because postgres requires certain super user permissions to use "ALL TABLES".
+                            //      We should restrict this to the configured tables here.
+                            stmt.execute("CREATE PUBLICATION dbz_publication FOR ALL TABLES;");
+                        }
+                        else {
+                            LOGGER.warn(
```

Comment:
Should it really be warning? There are user monitoring not only errors but warnings too and this looks like a valid state.

Reply:
Good point, its likely better to be a trace log message.